### PR TITLE
Bump version to v0.6

### PR DESCRIPTION
**Pull request by qinsoon**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,47 @@
...
+
+Policies
+---
+* Added the Immix space, and related data structures.
+* Added `get_forwarded_object()` and `is_reachable()` to the space function table (SFT).
+* Improved the Malloc space (marksweep).
+  * It now supports parallel sweeping.
+  * It now uses side metadata for page and chunk bits.
+  * It now supports bulk check for live objects if mark bits are side metadata.
+* Fixed a bug that the side forwarding bits are not cleared for CopySpace.
```
Comment (k-sareen) [inline CHANGELOG.md:21]: I would suggest past tense: Fixed a bug where the side forwarding bits were not cleared for CopySpace.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,47 @@
...
+
+API
+---
+* Refactored to the metadata specs in the ObjectModel trait. Now each spec
```
Comment (k-sareen) [inline CHANGELOG.md:25]: There is an extra "to" at the start -- should be "Refactored the metadata specs ..."

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,47 @@
...
+* Added support for VM specific weak reference processing:
+  * added an API function `on_closure_end()` to add a callback so the binding will
+    be notified when a object transitive closure ends.
+  * added `vm_release()` and `process_weak_refs()` to the `Collection` trait.
+
+Misc
+---
+* Added a few new plan constraints.
+  * needs_log_bit: indicates whether a plan will use the global log bit.
+  * may_trace_duplicate_edges: indicates whether a plan may allow benign race and trace duplicate edges in a GC.
```
Comment (k-sareen) [inline CHANGELOG.md:37]: "may allow benign races"

**Review by k-sareen — APPROVED** [review]
LGTM other than the few suggestions. Thanks!

**Comment by qinsoon** [comment]
I have made changes as suggested.

**Pull request merged**